autoload/neomake/makers/ft/rust.vim

```diff
@@ -135,16 +135,19 @@ function! neomake#makers#ft#rust#CargoProcessOutput(context) abort
         endif
 
         let error = {'maker_name': 'cargo'}
+
         let code_dict = get(data, 'code', -1)
-        if code_dict is g:neomake#compat#json_null
-            if get(data, 'level', '') ==# 'warning'
-                let error.type = 'W'
+        if code_dict isnot g:neomake#compat#json_null
+                    \ && index(['E', 'W'], code_dict['code'][0]) != -1
+            let error.type = code_dict['code'][0]
+            let error.nr = str2nr(code_dict['code'][1:])
+        else
+            let level = get(data, 'level', -1)
+            if level != -1
+                let error.type = toupper(level[0])
             else
-                let error.type = 'E'
+                let error.type = 'W'
```

Comment:
This fallback is not covered with tests.
Is it a valid use case?  (i.e. is there `"code": null`, without `"level"`?)